### PR TITLE
[sanitizer_symbolizer] Use correct format strings for uptr

### PR DESCRIPTION
**Pull request by DimitryAndric**
When compiling the common sanitizer libraries, there are many warnings about format specifiers, similar to:

    compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp:31:32: warning: format specifies type 'void *' but the argument has type 'uptr' (aka 'unsigned long') [-Wformat]
       31 |   buffer->AppendF(kFormatData, DI->start);
          |                   ~~~~~~~~~~~  ^~~~~~~~~
    compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup_constants.h:33:46: note: format string is defined here
       33 | constexpr const char *kFormatData = "{{{data:%p}}}";
          |                                              ^~
          |                                              %lu
    compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp:46:43: warning: format specifies type 'void *' but the argument has type 'uptr' (aka 'unsigned long') [-Wformat]
       46 |   buffer->AppendF(kFormatFrame, frame_no, address);
          |                   ~~~~~~~~~~~~            ^~~~~~~
    compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup_constants.h:36:48: note: format string is defined here
       36 | constexpr const char *kFormatFrame = "{{{bt:%u:%p}}}";
          |                                                ^~
          |                                                %lu
    ...

This is because `uptr` is dependent on the platform, and can be either `unsigned long long`, `unsigned long`, or `unsigned int`.

To fix the warnings, define the correct printf modifier for `uptr` and `sptr` as `SANITIZER_PTR_MOD`, and use it in the various format strings in `sanitizer_symbolizer_markup_constants.h`.

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup_constants.h [hunk 1]
```diff
@@ -26,20 +26,23 @@ constexpr const char *kFormatDemangle = "{{{symbol:%s}}}";
 constexpr uptr kFormatDemangleMax = 1024;  // Arbitrary.
 
 // Function name or equivalent from PC location.
-constexpr const char *kFormatFunction = "{{{pc:%p}}}";
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup_constants.h:29]: if uptr used for pointer, it should be `%p` but printf arg should be caster to (void*)
if it 's uptr for size, then `%zx` (for hex), or `%zu` should be used

**Comment by DimitryAndric** [comment]
Casting seems uglier to me, though. On the other hand, it's difficult to make printf-like functions type-safe in any way...

File: compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp [hunk 2]
```diff
@@ -96,7 +97,7 @@ static void RenderModule(InternalScopedString *buffer,
   for (uptr i = 0; i < module.uuid_size(); i++)
     buildIdBuffer.AppendF("%02x", module.uuid()[i]);
 
-  buffer->AppendF(kFormatModule, moduleId, module.full_name(),
+  buffer->AppendF(kFormatModule, static_cast<int>(moduleId), module.full_name(),
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp:100]: here, I'd rather update
"{{{module:%zu:%s:elf:%s}}}";

File: compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp [hunk 3]
```diff
@@ -118,8 +119,11 @@ static void RenderMmaps(InternalScopedString *buffer,
     // module.base_address == dlpi_addr
     // range.beg == dlpi_addr + p_vaddr
     // relative address == p_vaddr == range.beg - module.base_address
-    buffer->AppendF(kFormatMmap, range.beg, range.end - range.beg, moduleId,
-                    accessBuffer.data(), range.beg - module.base_address());
+    buffer->AppendF(
+        kFormatMmap, reinterpret_cast<void *>(range.beg),
+        static_cast<unsigned int>(range.end - range.beg),
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp:124]: it's unlucky sanitizer convension, but uptr is equivalend to size_t.

`static_cast<unsigned int>(range.end - range.beg),` is uptr so size format `%z` should be used

So can you try to "{{{mmap:%p:0x%zx:load:%zu:%s:0x%zx}}}";

File: compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp [hunk 4]
```diff
@@ -119,11 +119,10 @@ static void RenderMmaps(InternalScopedString *buffer,
     // module.base_address == dlpi_addr
     // range.beg == dlpi_addr + p_vaddr
     // relative address == p_vaddr == range.beg - module.base_address
-    buffer->AppendF(
-        kFormatMmap, reinterpret_cast<void *>(range.beg),
-        static_cast<unsigned int>(range.end - range.beg),
-        static_cast<int>(moduleId), accessBuffer.data(),
-        static_cast<unsigned int>(range.beg - module.base_address()));
+    buffer->AppendF(kFormatMmap, reinterpret_cast<void *>(range.beg),
+                    static_cast<size_t>(range.end - range.beg),
```
Comment (vitalybuka) [inline compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_markup.cpp:123]: no need to cast size_t

I am saying that compiler-rt (sanitizers) use uptr to format as size_t.
It's all over the code, it should be OK to pass uptr into %zx.

**Comment by vitalybuka** [comment]
Thank you!

**Review by vitalybuka — APPROVED** [review]

**Pull request merged**